keps/sig-cli/2551-return-code-normalization/README.md

```diff
@@ -0,0 +1,652 @@
+  - [Dependencies](#dependencies)
+  - [Scalability](#scalability)
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Standardize the exit codes of kubectl when the command is successful or when it fails
+
+## Motivation
+
+kubectl is composed of several commands and subcommands, that can deal only with functions
+as simple as getting the current version of server and client, to more complex functions
+as getting differences between a local file and the current created resource on a Kubernetes
+cluster, or doing some deployment roll out.
+
+Additionally, some of those commands may call other external system commands (like an editor, 
+or a diff command, or even a plugin). 
+
+This brings a complexity about assessing:
+* Was the command executed correctly
+* If not, where was the problem? On the server side? Was the certificate expired? Or the resource does not exists?
+* Was that a problem on the client side? Was that a problem of an external system command?
+
+We have a number of issues across the repo, as [kubernetes #99354](https://github.com/kubernetes/kubernetes/issues/99354),
+[kubectl #847](https://github.com/kubernetes/kubectl/issues/847), 
+[kubernetes #73056](https://github.com/kubernetes/kubernetes/issues/73056), 
+[kubernetes #39767](https://github.com/kubernetes/kubernetes/issues/39767) and 
+[kubernetes #26424](https://github.com/kubernetes/kubernetes/issues/26424) that points
+to this specific problem.
```

Comment:
Maybe a few word summary + link would be nice?

Reply:
@rikatz will tackle on this